packages/dom/src/dom/is-number-input.js

```diff
@@ -17,6 +17,6 @@ export default function isNumberInput( node ) {
 	return (
 		isHTMLInputElement( node ) &&
 		node.type === 'number' &&
-		!! node.valueAsNumber
+		! Number.isNaN( node.valueAsNumber )
```

Comment:
I was just thinking about this one, and I have no idea why the initial implementation even cared about inspecting the value. I'm not sure what this tells us. Is `<input type="number">` somehow _not_ a number input? I believe that it is, even if it has no value yet. In other words, I'm tempted to fix this issue by simply removing the `!! node.valueAsNumber` condition. What do you think?

Reply:
I'm not sure but it [seems to be connected](https://github.com/WordPress/gutenberg/pull/21457) with your related PR about handling number inputs :) . I'll close this one and see how to change/handle it in your PR.